src/dronevis/models/gesture_recognition.py

```diff
@@ -0,0 +1,202 @@
+from typing import Union
+import time
+import cv2
+import mediapipe as mp
+import numpy as np
+import torch
+from torch.autograd import Variable
+import torch.nn as nn
+import torch.nn.functional as F
+from torch.optim import *
+import itertools
+import copy
+import os
+from os.path import isfile, join
+
+from dronevis.abstract import CVModel
+from dronevis.utils.general import write_fps
+
+Keypoint_classifier_model_path = join(os.getcwd(),r'src\dronevis\models\hand_keypoint_model_weights.pth')
+device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
+gesture_labels_dict = {"Down":0,"Forward":1, "Left":2, "Right":3, "Stop":4, "Up":5}
+
+class GestureRecognition(CVModel):
+    """Gesture Recognition class with mediapipe
+
+    This class inherits from base class ``CVModel``, and implements
+    its abstract methods for code integrity.
+    """
+
+    def __init__(self, min_detection_confidence: float = 0.5, min_tracking_confidence: float=0.5) -> None:
+        """Construct model instance
+
+        Args:
+            min_detection_confidence(float, optional): threshold for detection
+            min_tracking_confidence(float, optional): threshold for tracking,
+            **input is a probability [0, 1]**.
+            Defaults to 0.5.
+        """
+        assert 0.0 <= min_detection_confidence <= 1, "Detection confidence must be a score between 0 and 1"
+        assert isinstance(min_detection_confidence, (int, float)), "Confidence must be a number"
+        
+        assert 0.0 <= min_tracking_confidence <= 1, "Tracking confidence must be a score between 0 and 1"
+        assert isinstance(min_tracking_confidence, (int, float)), "Tracking must be a number"
+
+        self.min_detection_confidence = min_detection_confidence
+        self.min_tracking_confidence = min_tracking_confidence
+
+        self.mp_drawing = mp.solutions.drawing_utils # type: ignore
+        self.mp_drawing_styles = mp.solutions.drawing_styles  # type: ignore
+        self.mp_hands = mp.solutions.hands  # type: ignore
+    
+    def load_model(self) -> None:
+        """Load model from memory"""
+        self.keypoints_classifier_model = Keypoint_classifier()
+        self.keypoints_classifier_model = self.keypoints_classifier_model.to(device)
+        self.keypoints_classifier_model.load_state_dict(torch.load(Keypoint_classifier_model_path))
+        self.keypoints_classifier_model.double()
+
+    def transform_img(self, image: np.ndarray) -> np.ndarray:
+        """Idle transformation of the image"""
+        return image
+    
+    
+    def predict(self, image: np.ndarray) -> np.ndarray:
+        """Run model inference on input image and output gesture keypoints and name
+
+        Args:
+            img (np.array): input image (assumed to be non-transformed)
+
+        Returns:
+            np.array: output image with keypoints drawn and gesture label recognized
+        """
+        image.flags.writeable = False
```

Comment:
Write a comment about what you are doing here, as it is NOT clear to the reader.